scintillometry/channelize/tests/test_channelize.py

```diff
@@ -0,0 +1,59 @@
+# Licensed under the GPLv3 - see LICENSE
+
+import numpy as np
+import astropy.units as u
+import pytest
+
+from .. import ChannelizeTask
+
+from baseband import vdif
+from baseband.data import SAMPLE_VDIF
+
+
+class TestChannelize(object):
+    """Test channelization using Baseband's sample VDIF file."""
+
+    def setup(self):
+        """Pre-calculate channelized data."""
+        self.n = 1024
+
+        with vdif.open(SAMPLE_VDIF) as fh:
+            self.ref_start_time = fh.start_time
+            self.ref_sample_rate = fh.sample_rate
+            data = fh.read()
+
+        last_sample = self.n * (data.shape[0] // self.n)
+        self.ref_data = np.fft.rfft(
+            data[:last_sample].reshape((-1, self.n) + data.shape[1:]),
+            axis=1).astype('complex64')
+
+    def test_channelizetask(self):
```

Comment:
Maybe check with both `pyfftw` and `numpy`? Easiest might be to use `pyfftw` if available.

Reply:
Fair enough - though obviously the test setup could use the same FFT routine. The real question here is testing the setting of the FFT. Anyway, am happy to leave that out for now.